src/code/z_camera.c

```diff
@@ -1,3070 +1,6778 @@
+    Vec3f atTarget;
     f32 sp38; // unused
     f32 temp_ret;
-    PosRot* temp_s1;
+    PosRot* playerPosRot = &camera->playerPosRot;
 
-    sp48.y = Player_GetCameraYOffset(camera->player) + arg2;
-    sp48.x = 0.0f;
-    sp48.z = 0.0f;
+    posOffsetTarget.y = Player_GetHeight(camera->player) + yExtra;
+    posOffsetTarget.x = 0.0f;
+    posOffsetTarget.z = 0.0f;
 
     temp_ret = Math_Sins(arg3);
-    temp_s1 = &camera->playerPosRot;
 
     if (temp_ret < 0.0f) {
-        phi_f2 = Math_Coss(temp_s1->rot.y - arg1->yaw);
+        phi_f2 = Math_Coss(playerPosRot->rot.y - eyeAtDir->yaw);
     } else {
-        phi_f2 = -Math_Coss(temp_s1->rot.y - arg1->yaw);
+        phi_f2 = -Math_Coss(playerPosRot->rot.y - eyeAtDir->yaw);
     }
 
-    sp48.y -= temp_ret * phi_f2 * OREG(9);
-    func_80043A3C(&sp48, &camera->unk_E4, camera->unk_CC.y, camera->unk_CC.x, 0.1f);
+    posOffsetTarget.y -= temp_ret * phi_f2 * OREG(9);
+    Camera_LERPCeilVec3f(&posOffsetTarget, &camera->posOffset, camera->yOffsetUpdateRate, camera->xzOffsetUpdateRate,
+                         0.1f);
+
+    atTarget.x = playerPosRot->pos.x + camera->posOffset.x;
+    atTarget.y = playerPosRot->pos.y + camera->posOffset.y;
+    atTarget.z = playerPosRot->pos.z + camera->posOffset.z;
+    Camera_LERPCeilVec3f(&atTarget, &camera->at, camera->atLERPStepScale, camera->atLERPStepScale, 0.2f);
 
-    sp3C.x = temp_s1->pos.x + camera->unk_E4.x;
-    sp3C.y = temp_s1->pos.y + camera->unk_E4.y;
-    sp3C.z = temp_s1->pos.z + camera->unk_E4.z;
-    func_80043A3C(&sp3C, &camera->at, camera->unk_100, camera->unk_100, 0.2f);
+    return 1;
+}
+
+/**
+ * Adjusts the camera's at position for Camera_Parallel1
+ */
+s32 Camera_CalcAtForParallel(Camera* camera, VecSph* arg1, f32 arg2, f32* arg3, s16 arg4) {
+    Vec3f* at = &camera->at;
+    Vec3f posOffsetTarget;
+    Vec3f atTarget;
+    Vec3f* eye = &camera->eye;
+    PosRot* playerPosRot = &camera->playerPosRot;
+    f32 temp_f2;
+    f32 phi_f16;
+    f32 sp54;
+    f32 phi_f20;
+    f32 temp_f0_4;
+
+    temp_f0_4 = Player_GetHeight(camera->player);
+    posOffsetTarget.x = 0.0f;
+    posOffsetTarget.y = temp_f0_4 + arg2;
+    posOffsetTarget.z = 0.0f;
+
+    if (PREG(76) && arg4) {
+        posOffsetTarget.y -= Camera_CalcSlopeYAdj(&camera->floorNorm, playerPosRot->rot.y, arg1->yaw, OREG(9));
+    }
 
+    if (camera->playerGroundY == camera->playerPosRot.pos.y || camera->player->actor.gravity > -0.1f ||
+        camera->player->stateFlags1 & 0x200000) {
+        *arg3 = Camera_LERPCeilF(playerPosRot->pos.y, *arg3, PCT(OREG(43)), 0.1f);
+        phi_f20 = playerPosRot->pos.y - *arg3;
+        posOffsetTarget.y -= phi_f20;
+        Camera_LERPCeilVec3f(&posOffsetTarget, &camera->posOffset, camera->yOffsetUpdateRate,
+                             camera->xzOffsetUpdateRate, 0.1f);
+    } else {
+        // this condition can basically never be met, due to the decision to call
+        // this function in Camera_Parallel1
```

Comment:
@bug tag? not sure if this qualifies
what does "basically" mean? are there some niche situations where it will be met?

Reply:
I'm just going to remove this comment, because this function could used for modding purposes, but in the current implementation, the previous if statement 
`if (camera->playerGroundY == camera->playerPosRot.pos.y || camera->player->actor.gravity > -0.1f ||
        camera->player->stateFlags1 & 0x200000) {` will always be true, because the function is only ever called based on the same if statement.